src/realm/sync/noinst/client_reset.cpp

```diff
@@ -1950,18 +1951,31 @@ ClientResyncMode reset_precheck_guard(Transaction& wt, ClientResyncMode mode, ut
                                                      previous_reset->type, previous_reset->time, mode));
         }
     }
+    if (!recovery_is_allowed) {
+        if (mode == ClientResyncMode::Recover) {
+            throw ClientResetFailed(
```

Comment:
I know this has probably been bike shedded to death, but what is a user supposed to do when this happens? I guess the only backup they could do here is to set a manual client reset handler, but I don't see where we allow falling back to that.

Reply:
Any exception thrown during the client reset operation is caught higher up by client_impl_base which generates a `ClientError::auto_client_reset_failure` error which will trigger a manual reset by the session error handler. TBH, there are not a lot of good options at that point, but I wanted to give the option of creating a fatal error (potentially but unlikely handled by the user manually) instead of just blindly opting for discarding local changes. Presumably some apps will not like the idea of discarding data when recovery is not allowed by the server so this is the path for them.